app/views/shared/_dislike.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= link_to(like_path(post_id: @post, like_id: @post.likes.find_by(user_id: current_user.id).id), id: "dislike_#{@post.likes.find_by(user_id: current_user.id).id}", method: :delete, data: {remote: true}, class: "dislike") do%>
```

Comment:
- [ ] There should be no application logic on the views or helpers. -- views and helper files are for view-related logic but not application-related logic.
Application logic usually hit the database to grab information like your methods here `find_by` and it's place is not in the views. Views and View helpers are only responsible for displaying information that became available to them through controllers, they supposed to have no knowledge on how to get these data. As the name `View` suggests, it's for just **view**ing.